python313: cython and pyliblo with liblo throws cython-0.29.37.1 not supported for interpreter python3.13 error

### Nixpkgs version

- Unstable (25.05)

### Describe the bug

When using [pyliblo](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/pyliblo/default.nix#L39) with Python 3.13, a development shell build fails due to a Cython dependency for the liblo library.

### Steps to reproduce

Use the following `flake.nix` to create a nix development shell using `nix develop .#default`.

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    flake-utils.url = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, flake-utils, ... }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        pkgs = import nixpkgs {
          inherit system;
        };
        customPython = pkgs.python313.withPackages (ps: with ps; [
          #cython # commenting out this line does not help either
          pkgs.liblo
          pyliblo
        ]);
      in {
        devShells = {
          default = pkgs.mkShell {
            buildInputs = [
              customPython
            ];
          };
        };
      });
}
```


### Expected behaviour

Python 3.13 should build with pyliblo with a current Cython version

### Screenshots

_No response_

### Relevant log output

```console
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/stdenv/generic/make-derivation.nix:375:7

       … while evaluating attribute 'buildInputs' of derivation 'nix-shell'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/stdenv/generic/make-derivation.nix:422:7:
          421|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          422|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          423|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3-3.13.2-env'
         whose name attribute is located at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/stdenv/generic/make-derivation.nix:375:7

       … while evaluating attribute 'passAsFile' of derivation 'python3-3.13.2-env'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/trivial-builders/default.nix:64:9:
           63|         inherit buildCommand name;
           64|         passAsFile = [ "buildCommand" ]
             |         ^
           65|           ++ (derivationArgs.passAsFile or [ ]);

       … while evaluating the attribute 'passAsFile'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:113:9:
          112|         # XXX: The size is somewhat arbitrary
          113|         passAsFile = if builtins.stringLength pkgs >= 128 * 1024 then [ "pkgs" ] else [ ];
             |         ^
          114|       }

       … while evaluating a branch condition
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:113:22:
          112|         # XXX: The size is somewhat arbitrary
          113|         passAsFile = if builtins.stringLength pkgs >= 128 * 1024 then [ "pkgs" ] else [ ];
             |                      ^
          114|       }

       … in the argument of the not operator
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:113:52:
          112|         # XXX: The size is somewhat arbitrary
          113|         passAsFile = if builtins.stringLength pkgs >= 128 * 1024 then [ "pkgs" ] else [ ];
             |                                                    ^
          114|       }

       … while calling the 'lessThan' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:113:52:
          112|         # XXX: The size is somewhat arbitrary
          113|         passAsFile = if builtins.stringLength pkgs >= 128 * 1024 then [ "pkgs" ] else [ ];
             |                                                    ^
          114|       }

       … while calling the 'stringLength' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:113:25:
          112|         # XXX: The size is somewhat arbitrary
          113|         passAsFile = if builtins.stringLength pkgs >= 128 * 1024 then [ "pkgs" ] else [ ];
             |                         ^
          114|       }

       … while calling the 'toJSON' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:108:16:
          107|           ;
          108|         pkgs = builtins.toJSON chosenOutputs;
             |                ^
          109|         extraPathsFrom = lib.optional includeClosures (writeClosure pathsForClosure);

       … while calling the 'map' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/build-support/buildenv/default.nix:67:21:
           66|   let
           67|     chosenOutputs = map (drv: {
             |                     ^
           68|       paths =

       … from call site
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/development/interpreters/python/wrapper.nix:20:13:
           19|   env = let
           20|     paths = requiredPythonModules (extraLibs ++ [ python ] ) ;
             |             ^
           21|     pythonPath = "${placeholder "out"}/${python.sitePackages}";

       … while calling 'requiredPythonModules'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/development/interpreters/python/python-packages-base.nix:65:5:
           64|   requiredPythonModules =
           65|     drvs:
             |     ^
           66|     let

       … while calling the 'foldl'' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/development/interpreters/python/python-packages-base.nix:69:5:
           68|     in
           69|     lib.unique (
             |     ^
           70|       [ python ] ++ modules ++ lib.concatLists (lib.catAttrs "requiredPythonModules" modules)

       … while calling anonymous lambda
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/lists.nix:1847:25:
         1846|   */
         1847|   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
             |                         ^
         1848|

       … while evaluating a branch condition
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/lists.nix:1847:28:
         1846|   */
         1847|   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
             |                            ^
         1848|

       … while calling the 'elem' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/lists.nix:1847:31:
         1846|   */
         1847|   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
             |                               ^
         1848|

       … while evaluating the attribute 'outPath'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/customisation.nix:421:7:
          420|         drv.drvPath;
          421|       outPath =
             |       ^
          422|         assert condition;

       … while evaluating the attribute 'outPath'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/customisation.nix:421:7:
          420|         drv.drvPath;
          421|       outPath =
             |       ^
          422|         assert condition;

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:44:19:
           43|       value = commonAttrs // {
           44|         outPath = builtins.getAttr outputName strict;
             |                   ^
           45|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'python3.13-pyliblo-0.10.0'
         whose name attribute is located at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/stdenv/generic/make-derivation.nix:375:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'python3.13-pyliblo-0.10.0'
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/stdenv/generic/make-derivation.nix:419:7:
          418|       depsBuildBuild              = elemAt (elemAt dependencies 0) 0;
          419|       nativeBuildInputs           = elemAt (elemAt dependencies 0) 1;
             |       ^
          420|       depsBuildTarget             = elemAt (elemAt dependencies 0) 2;

       … in the condition of the assert statement
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/lib/customisation.nix:401:15:
          400|             outPath =
          401|               assert condition;
             |               ^
          402|               drv.${outputName}.outPath;

       … in the right operand of the IMPL (->) operator
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/development/interpreters/python/mk-python-derivation.nix:467:7:
          466|       drv.disabled
          467|       -> throw "${removePrefix namePrefix drv.name} not supported for interpreter ${python.executable}"
             |       ^
          468|     ) { } (toPythonModule drv);

       … while calling the 'throw' builtin
         at /nix/store/pqgpwakx0m4r75zg34954w5mwi5shg89-source/pkgs/development/interpreters/python/mk-python-derivation.nix:467:10:
          466|       drv.disabled
          467|       -> throw "${removePrefix namePrefix drv.name} not supported for interpreter ${python.executable}"
             |          ^
          468|     ) { } (toPythonModule drv);

       error: cython-0.29.37.1 not supported for interpreter python3.13
```

### Additional context

Based on the logs I believe this is a versioning problem as the current Cython version is 3.0.12, and nixos-unstable currently has 3.0.11. However, the log seems to use a different versioning system as it states 0.29.37.1 which is confusing, I'm not sure where that version number is from.

I've found a similar issue #332407, however, the author of these issue has a rather large nixos configuration that installs Python/Cython somewhere and is difficult to narrow down. Hopefully, this shell file can provide a better starting point for debugging.

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.17, NixOS, 25.05 (Warbler), 25.05.20250305.d69ab0d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/b19xi89pgm1arig3l9xqn1h81zjd1j91-source`
```

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

```
cython> running build_ext
cython> building 'Cython.Plex.Scanners' extension
cython> creating build/temp.linux-x86_64-cpython-313/build/source/Cython/Plex
cython> gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/h8wi6kwl1cywxr3f22836vgrrb11fvlb-libxcrypt-4.4.38/include -fPIC -I/nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13 -c /build/source/Cython/Plex/Scanners.c -o build/temp.linux-x86_64-cpython-313/build/source/Cython/Plex/Scanners.o
cython> /nix/store/ki4if6b0w5bqv8dc5lrjp8xm7wjy9dlf-bash-5.2p37/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8): No such file or directory
cython> /build/source/Cython/Plex/Scanners.c:843:1: warning: 'Py_UNICODE' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
cython>   843 | static CYTHON_INLINE size_t __Pyx_Py_UNICODE_strlen(const Py_UNICODE *u) {
cython>       | ^~~~~~
cython> /build/source/Cython/Plex/Scanners.c: In function '__Pyx_Py_UNICODE_strlen':
cython> /build/source/Cython/Plex/Scanners.c:844:5: warning: 'Py_UNICODE' is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
cython>   844 |     const Py_UNICODE *u_end = u;
cython>       |     ^~~~~
cython> /build/source/Cython/Plex/Scanners.c: In function '__Pyx_PyInt_As_long':
cython> /build/source/Cython/Plex/Scanners.c:10821:27: error: too few arguments to function '_PyLong_AsByteArray'
cython> 10821 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
cython>       |                           ^~~~~~~~~~~~~~~~~~~
cython> In file included from /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/longobject.h:107,
cython>                  from /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/Python.h:81,
cython>                  from /build/source/Cython/Plex/Scanners.c:21:
cython> /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/cpython/longobject.h:111:17: note: declared here
cython>   111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
cython>       |                 ^~~~~~~~~~~~~~~~~~~
cython> /build/source/Cython/Plex/Scanners.c: In function '__Pyx_PyInt_As_int':
cython> /build/source/Cython/Plex/Scanners.c:11017:27: error: too few arguments to function '_PyLong_AsByteArray'
cython> 11017 |                 int ret = _PyLong_AsByteArray((PyLongObject *)v,
cython>       |                           ^~~~~~~~~~~~~~~~~~~
cython> /nix/store/1031nlccnpdp7ljbjm4ja0q7gqadsl3m-python3-3.13.2/include/python3.13/cpython/longobject.h:111:17: note: declared here
cython>   111 | PyAPI_FUNC(int) _PyLong_AsByteArray(PyLongObject* v,
cython>       |                 ^~~~~~~~~~~~~~~~~~~
cython> error: command '/nix/store/gj9lra51hwhxnhz05jqk5lh03wipamv0-gcc-wrapper-14-20241116/bin/gcc' failed with exit code 1
cython> 
cython> ERROR Backend subprocess exited when trying to invoke build_wheel
```

```
pyliblo> Executing setuptoolsBuildPhase
pyliblo> setup.py build flags: ''
pyliblo> /nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py:270: UserWarning: Unknown distribution option: 'test_suite'
pyliblo>   warnings.warn(msg)
pyliblo> running bdist_wheel
pyliblo> running build
pyliblo> running build_ext
pyliblo> [1/1] Cythonizing src/liblo.pyx
pyliblo> /nix/store/xcbj7giv259kv23j0baqkhs7qf6ndmz0-python3.12-cython-3.0.11-1/lib/python3.12/site-packages/Cython/Compiler/Main.py:381: FutureWarning: Cython directive 'language_level' not set, using '3str' for now (Py3). This has changed from earlier releases! File: /build/pyliblo-0.10.0/src/liblo.pxd
pyliblo>   tree = Parsing.p_module(s, pxd, full_module_name)
pyliblo> performance hint: src/liblo.pyx:286:5: Exception check on '_err_handler' will always require the GIL to be acquired.
pyliblo> Possible solutions:
pyliblo>         1. Declare '_err_handler' as 'noexcept' if you control the definition and you're sure you don't want the function to raise exceptions.
pyliblo>         2. Use an 'int' return type on '_err_handler' to allow an error code to be returned.
pyliblo> 
pyliblo> Error compiling Cython file:
pyliblo> ------------------------------------------------------------
pyliblo> ...
pyliblo>         # class)
pyliblo>         cb = struct(func=_weakref_method(func), user_data=user_data)
pyliblo>         # keep a reference to the callback data around
pyliblo>         self._keep_refs.append(cb)
pyliblo> 
pyliblo>         lo_server_add_method(self._server, p, t, _msg_callback, <void*>cb)
pyliblo>                                                  ^
pyliblo> ------------------------------------------------------------
pyliblo> 
pyliblo> src/liblo.pyx:457:49: Cannot assign type 'int (const_char *, const_char *, lo_arg **, int, lo_message, void *) except? -1 nogil' to 'lo_method_handler' (alias of 'int (*)(const_char *, const_char *, lo_arg **, int, lo_message, void *) noexcept'). Exception values are incompatible. Suggest adding 'noexcept' to the type of '_msg_callback'.
pyliblo> 
pyliblo> Error compiling Cython file:
pyliblo> ------------------------------------------------------------
pyliblo> ...
pyliblo>         cb_data = struct(start_func=_weakref_method(start_handler),
pyliblo>                          end_func=_weakref_method(end_handler),
pyliblo>                          user_data=user_data)
pyliblo>         self._keep_refs.append(cb_data)
pyliblo> 
pyliblo>         lo_server_add_bundle_handlers(self._server, _bundle_start_callback,
pyliblo>                                                     ^
pyliblo> ------------------------------------------------------------
pyliblo> 
pyliblo> src/liblo.pyx:512:52: Cannot assign type 'int (lo_timetag, void *) except? -1 nogil' to 'lo_bundle_start_handler' (alias of 'int (*)(lo_timetag, void *) noexcept'). Exception values are incompatible. Suggest adding 'noexcept' to the type of '_bundle_start_callback'.
pyliblo> 
pyliblo> Error compiling Cython file:
pyliblo> ------------------------------------------------------------
pyliblo> ...
pyliblo>                          end_func=_weakref_method(end_handler),
pyliblo>                          user_data=user_data)
pyliblo>         self._keep_refs.append(cb_data)
pyliblo> 
pyliblo>         lo_server_add_bundle_handlers(self._server, _bundle_start_callback,
pyliblo>                                       _bundle_end_callback, <void*>cb_data)
pyliblo>                                       ^
pyliblo> ------------------------------------------------------------
pyliblo> 
pyliblo> src/liblo.pyx:513:38: Cannot assign type 'int (void *) except? -1 nogil' to 'lo_bundle_end_handler' (alias of 'int (*)(void *) noexcept'). Exception values are incompatible. Suggest adding 'noexcept' to the type of '_bundle_end_callback'.
pyliblo> 
pyliblo> Error compiling Cython file:
pyliblo> ------------------------------------------------------------
pyliblo> ...
pyliblo>         else:
pyliblo>             cs = NULL
pyliblo> 
pyliblo>         global __exception
pyliblo>         __exception = None
pyliblo>         self._server = lo_server_new_with_proto(cs, proto, _err_handler)
pyliblo>                                                            ^
pyliblo> ------------------------------------------------------------
pyliblo> 
pyliblo> src/liblo.pyx:600:59: Cannot assign type 'void (int, const_char *, const_char *) except * nogil' to 'lo_err_handler' (alias of 'void (*)(int, const_char *, const_char *) noexcept'). Exception values are incompatible. Suggest adding 'noexcept' to the type of '_err_handler'.
pyliblo> 
pyliblo> Error compiling Cython file:
pyliblo> ------------------------------------------------------------
pyliblo> ...
pyliblo>         # make sure python can handle threading
pyliblo>         PyEval_InitThreads()
pyliblo> 
pyliblo>         global __exception
pyliblo>         __exception = None
pyliblo>         self._server_thread = lo_server_thread_new_with_proto(cs, proto, _err_handler)
pyliblo>                                                                          ^
pyliblo> ------------------------------------------------------------
pyliblo> 
pyliblo> src/liblo.pyx:699:73: Cannot assign type 'void (int, const_char *, const_char *) except * nogil' to 'lo_err_handler' (alias of 'void (*)(int, const_char *, const_char *) noexcept'). Exception values are incompatible. Suggest adding 'noexcept' to the type of '_err_handler'.
pyliblo> Traceback (most recent call last):
pyliblo>   File "/build/pyliblo-0.10.0/nix_run_setup", line 8, in <module>
pyliblo>     exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
pyliblo>   File "setup.py", line 54, in <module>
pyliblo>     setup(
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
pyliblo>     return distutils.core.setup(**attrs)
pyliblo>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 186, in setup
pyliblo>     return run_commands(dist)
pyliblo>            ^^^^^^^^^^^^^^^^^^
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
pyliblo>     dist.run_commands()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 983, in run_commands
pyliblo>     self.run_command(cmd)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/dist.py", line 999, in run_command
pyliblo>     super().run_command(command)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
pyliblo>     cmd_obj.run()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/command/bdist_wheel.py", line 379, in run
pyliblo>     self.run_command("build")
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
pyliblo>     self.distribution.run_command(command)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/dist.py", line 999, in run_command
pyliblo>     super().run_command(command)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
pyliblo>     cmd_obj.run()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/command/build.py", line 136, in run
pyliblo>     self.run_command(cmd_name)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 339, in run_command
pyliblo>     self.distribution.run_command(command)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/dist.py", line 999, in run_command
pyliblo>     super().run_command(command)
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
pyliblo>     cmd_obj.run()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 365, in run
pyliblo>     self.build_extensions()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 481, in build_extensions
pyliblo>     self._build_extensions_serial()
pyliblo>   File "/nix/store/chyqbfba0snalr78365hpd8imklspb4k-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/command/build_ext.py", line 507, in _build_extensions_serial
pyliblo>     self.build_extension(ext)
pyliblo>   File "/nix/store/xcbj7giv259kv23j0baqkhs7qf6ndmz0-python3.12-cython-3.0.11-1/lib/python3.12/site-packages/Cython/Distutils/build_ext.py", line 130, in build_extension
pyliblo>     new_ext = cythonize(
pyliblo>               ^^^^^^^^^^
pyliblo>   File "/nix/store/xcbj7giv259kv23j0baqkhs7qf6ndmz0-python3.12-cython-3.0.11-1/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 1154, in cythonize
pyliblo>     cythonize_one(*args)
pyliblo>   File "/nix/store/xcbj7giv259kv23j0baqkhs7qf6ndmz0-python3.12-cython-3.0.11-1/lib/python3.12/site-packages/Cython/Build/Dependencies.py", line 1321, in cythonize_one
pyliblo>     raise CompileError(None, pyx_file)
pyliblo> Cython.Compiler.Errors.CompileError: src/liblo.pyx
```

--- a/pkgs/development/python-modules/fugashi/default.nix
+++ b/pkgs/development/python-modules/fugashi/default.nix
@@ -4,7 +4,7 @@
   pythonOlder,
   pytestCheckHook,
   buildPythonPackage,
-  cython_0,
+  cython,
   mecab,
   setuptools-scm,
   ipadic,
@@ -14,19 +14,19 @@
 
 buildPythonPackage rec {
   pname = "fugashi";
-  version = "1.3.0";
-  format = "setuptools";
-  disabled = pythonOlder "3.7";
+  version = "1.5.1";
+  format = "pyproject";
+  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "polm";
     repo = "fugashi";
     tag = "v${version}";
-    hash = "sha256-4i7Q+TtXTQNSJ1EIcS8KHrVPdCJAgZh86Y6lB8772XU=";
+    hash = "sha256-rkQskRz7lgVBrqBeyj9kWO2/7POrZ0TaM+Z7mhpZLvM=";
   };
 
   nativeBuildInputs = [
-    cython_0
+    cython
     mecab
     setuptools-scm
   ];

